cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueClientFactory.java

```diff
@@ -42,35 +44,37 @@
   private final HBaseQueueAdmin queueAdmin;
   private final HBaseStreamAdmin streamAdmin;
   private final HBaseQueueUtil queueUtil;
+  private final HBaseTableUtil hBaseTableUtil;
 
   @Inject
-  public HBaseQueueClientFactory(CConfiguration cConf, Configuration hConf,
+  public HBaseQueueClientFactory(CConfiguration cConf, Configuration hConf, HBaseTableUtil hBaseTableUtil,
                                  QueueAdmin queueAdmin, HBaseStreamAdmin streamAdmin) {
     this.cConf = cConf;
     this.hConf = hConf;
     this.queueAdmin = (HBaseQueueAdmin) queueAdmin;
     this.streamAdmin = streamAdmin;
     this.queueUtil = new HBaseQueueUtilFactory().get();
+    this.hBaseTableUtil = hBaseTableUtil;
   }
 
   // for testing only
-  String getTableName(QueueName queueName) {
-    return (queueName.isStream() ? streamAdmin : queueAdmin).getActualTableName(queueName);
+  TableId getTableId(QueueName queueName) {
```

Comment:
Isn't there a difference - its not just non-private for the sake of testing, it exists only for testing?
`@VisibleForTesting` gives the impression that its used elsewhere, but non-private for testing reasons.